### PR TITLE
Add goeland module

### PR DESCRIPTION
**Pull request by sweenu**
###### Description of changes
Add a module for the [goeland](https://github.com/slurdge/goeland) program.

I have been using the module myself without any issue.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/misc/ids.nix [hunk 1]
```diff
@@ -666,6 +667,7 @@ in
       rstudio-server = 324;
       localtimed = 325;
       automatic-timezoned = 326;
+      goeland = 327;
```
Comment (SuperSandro2000) [inline nixos/modules/misc/ids.nix:670]: We no longer want to add new IDs here unless absolutely necessary. Can we just use users.users?

File: nixos/modules/services/mail/goeland.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    schedule = mkOption {
+      type = types.str;
+      default = "12h";
+      example = "Mon, 00:00:00";
+      description = mdDoc "How often to run goeland, in systemd time format";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.goeland.settings.database = mkDefault "/var/lib/goeland/goeland.db";
```
Comment (SuperSandro2000) [inline nixos/modules/services/mail/goeland.nix:32]: We should create the directories with tmpfiles.d

File: nixos/modules/services/mail/goeland.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      };
+      startAt = cfg.schedule;
+    };
+
+    users.users.goeland = {
+      description = "goeland user";
+      uid = config.ids.uids.goeland;
+      group = "goeland";
+    };
+    users.groups.goeland.gid = config.ids.gids.goeland;
```
Comment (SuperSandro2000) [inline nixos/modules/services/mail/goeland.nix:47]:
```suggestion
      group = "goeland";
    };
    users.groups.goeland = {};
```

File: nixos/modules/services/mail/goeland.nix (hunk 2 above)
Comment (sweenu) [reply nixos/modules/services/mail/goeland.nix:32]: It is a sqlite db that is supposed to be kept across reboots, is it still a good candidate for tmpfiles.d?

File: nixos/modules/misc/ids.nix (hunk 1 above)
Comment (sweenu) [reply nixos/modules/misc/ids.nix]: Oh okay, removed 👍🏽

File: nixos/modules/services/mail/goeland.nix (hunk 2 above)
Comment (SuperSandro2000) [reply nixos/modules/services/mail/goeland.nix:32]: tmpfiles.d can be used to create directories that hold data across reboots.

Comment (sweenu) [reply nixos/modules/services/mail/goeland.nix:34]: Okay. Added it to the rules 👍🏽

File: nixos/modules/services/mail/goeland.nix [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      type = types.str;
+      default = "12h";
+      example = "Mon, 00:00:00";
+      description = mdDoc "How often to run goeland, in systemd time format";
+    };
+  };
+
+  config =
+    let
+      goelandDirectory = "/var/lib/goeland";
```
Comment (SuperSandro2000) [inline nixos/modules/services/mail/goeland.nix:33]: Sorry that I noticed this so late but this should really be an option.

**Review by SuperSandro2000 — COMMENTED** [review]
otherwise LGTM

File: nixos/modules/services/mail/goeland.nix (hunk 4 above)
Comment (sweenu) [reply nixos/modules/services/mail/goeland.nix]: No worries, I added the option 👍🏽

**Comment by piegamesde** [comment]
Sadly people cannot be un-pinged on GitHub. So to avoid further spam, the only solution is to close the pull request and open a new one.

**Pull request closed without merging**